platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/Capture.kt

```diff
@@ -37,13 +50,18 @@ internal sealed class LoggerState {
      * The logger has been successfully started and is ready for use. Subsequent attempts to start the logger will be ignored.
      */
     class Started(
-        val logger: LoggerImpl,
+        internal val logger: ILogger,
     ) : LoggerState()
 
     /**
      * An attempt to start the logger was made but failed. Subsequent attempts to start the logger will be ignored.
      */
-    data object StartFailure : LoggerState()
+    class StartFailure(
+        /**
+         * The reason for Capture.Logger.start() failure
+         */
+        val throwable: Throwable,
```

Comment:
for functional programming style I wouldn't return throwables (since they can't really be caught). Take a look at our existing `sealed class Error` for a better way to return async errors